lux-lib/src/lockfile/mod.rs

```diff
@@ -221,14 +247,34 @@ pub(crate) enum RemotePackageSourceUrl {
 }
 
 // TODO(vhyrro): Move to `package/local.rs`
-#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, FromLua)]
 pub struct LocalPackage {
     pub(crate) spec: LocalPackageSpec,
     pub(crate) source: RemotePackageSource,
     pub(crate) source_url: Option<RemotePackageSourceUrl>,
     hashes: LocalPackageHashes,
 }
 
+impl UserData for LocalPackage {
+    fn add_methods<M: mlua::UserDataMethods<Self>>(methods: &mut M) {
+        methods.add_method("id", |_, this, _: ()| Ok(this.id()));
+        methods.add_method("name", |_, this, _: ()| Ok(this.name().clone()));
+        methods.add_method("version", |_, this, _: ()| Ok(this.version().clone()));
+        methods.add_method("pinned", |_, this, _: ()| Ok(this.pinned()));
+        methods.add_method("dependencies", |_, this, _: ()| {
+            Ok(this.spec.dependencies.clone())
+        });
+        methods.add_method("constraint", |_, this, _: ()| {
+            Ok(this.spec.constraint.clone())
+        });
+        methods.add_method("hashes", |_, this, _: ()| Ok(this.hashes.clone()));
+        methods.add_method("to_package", |_, this, _: ()| Ok(this.to_package()));
+        methods.add_method("into_package_req", |_, this, _: ()| {
+            Ok(this.clone().into_package_req())
+        });
```

Comment:
suggestion: since Lua doesn't have a concept of ownership, perhaps this should be

```suggestion
        methods.add_method("to_package_req", |_, this, _: ()| {
            Ok(this.clone().into_package_req())
        });
```